src/components/chart/state-history-chart-line.ts

```diff
@@ -84,6 +93,20 @@ class StateHistoryChartLine extends LitElement {
               display: true,
               text: this.unit,
             },
+            afterUpdate: (y) => {
+              if (this._yWidth !== Math.floor(y.width)) {
+                this._yWidth = Math.floor(y.width);
+                this.dispatchEvent(
+                  new CustomEvent("y-width-changed", {
```

Comment:
Use https://github.com/home-assistant/frontend/blob/dev/src/common/dom/fire_event.ts#L57

Reply:
I'm not sure I understand all the implications of this. I tried calling fireEvent and it compllains that y-width-changed is not valid for HASSDomEvents.

I tried copying some other code that had custom event type, and added:

```
declare global {
  interface HASSDomEvents {
    "y-width-changed": YWidthChangedEvent;
  }
}

interface YWidthChangedEvent {
  value: number,
  chartIndex: number;
}
```

But this makes the entire chart vanish, with no warnings or log messages... I'm not sure if I'm supposed to add this or not.